src/tickit/core/components/system_simulation.py

```diff
@@ -65,8 +72,33 @@ async def on_tick(self, time: SimTime, changes: Changes) -> None:
             changes (Changes): A mapping of changed component inputs and their new
                 values.
         """
-        output_changes, call_in = await self.scheduler.on_tick(time, changes)
-        await self.output(time, output_changes, call_in)
+        on_tick_task = asyncio.create_task(self.scheduler.on_tick(time, changes))
+        error_state = asyncio.create_task(self.scheduler.error.wait())
+
+        done, _ = await asyncio.wait(
+            [on_tick_task, error_state],
+            return_when=asyncio.tasks.FIRST_COMPLETED,
+        )
+        if error_state in done:
+            await self.state_producer.produce(
+                output_topic(self.name),
+                self.scheduler.component_error,
+            )
+
+        elif on_tick_task in done:
+            output_changes, call_in = on_tick_task.result()
+            await self.output(time, output_changes, call_in)
```

Comment:
The early return pattern would make the good path more easily read here, e.g.:

```suggestion
        if error_state in done:
            await self.state_producer.produce(
                output_topic(self.name),
                self.scheduler.component_error,
            )
            return

        output_changes, call_in = on_tick_task.result()
        await self.output(time, output_changes, call_in)
```
@callumforrester, what do you think?